libs/category/src/filters/validators/request-type-match.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  DaffCategoryFilterRequest,
+  DaffCategoryFilter,
+  DaffCategoryFilterToggleRequest,
+} from '../../models/public_api';
+import { DaffCategoryFilterRequestTypeMismatch } from '../errors/request-type-mismatch.error';
+
+/**
+ * Ensures that the types of the filter and the filter request match.
+ *
+ * @throws {@link DaffCategoryFilterRequestTypeMismatch} if the types do not match.
+ */
+export const daffCategoryValidateFilterRequestTypeMatch = (
+  request: DaffCategoryFilterRequest | DaffCategoryFilterToggleRequest,
+  filter: DaffCategoryFilter,
+): void => {
+  if (request.type !== filter.type) {
```

Comment:
I don't think this should assume that the filter will exist

Reply:
Do you think we should specifically check for a missing filter and throw an error or just optional chain `filter?.type`?

Reply:
I think just optional chaining is fine, since in the greater scheme of things the filter can't really be null here I think

Reply:
yeah and we do have a validator that checks for filter existence